lib/spack/spack/package_base.py

```diff
@@ -1909,23 +1877,39 @@ def do_test(self, dirty=False, externals=False):
                 )
                 return
 
-        kwargs = {"dirty": dirty, "fake": False, "context": "test", "externals": externals}
-        if tty.is_verbose():
-            kwargs["verbose"] = True
-        spack.build_environment.start_build_process(self, test_process, kwargs)
+        kwargs = {
+            "dirty": dirty,
+            "fake": False,
+            "context": "test",
+            "externals": externals,
+            "verbose": tty.is_verbose(),
+        }
+
+        self.tester.stand_alone_tests(kwargs)
 
+    # TODO (post-34236): Remove this deprecated method when eliminate test,
+    # TODO (post-34236): run_test, etc.
+    def _test_deprecated_warning(self):
```

Comment:
(Bug) This is what I meant a couple of comments above:
```suggestion
    @property
    def _test_deprecated_warning(self):
```

Reply:
Yes, I got that from your description.